scripts/svm/simpleDeposit.ts

```diff
@@ -83,9 +89,25 @@ async function depositV3(): Promise<void> {
     { property: "vault", value: vault.toString() },
   ]);
 
-  const tx = await (
+  const userTokenAccount = getAssociatedTokenAddressSync(inputToken, signer.publicKey);
+
+  const tokenDecimals = (await getMint(provider.connection, inputToken, undefined, TOKEN_PROGRAM_ID)).decimals;
+
+  // Delegate state PDA to pull depositor tokens.
+  const approveIx = await createApproveCheckedInstruction(
```

Comment:
cool. this is not too bad all things considered. did you test this on testnet with both token2022 and legacy token types?

Reply:
yes, here is old token [deposit](https://explorer.solana.com/tx/3x95k3WxgaqPWTQfUGaLzQr1udCat54oYtztUwJ31DqdG8TAyDq83dmXwNnqF1dzw7KtiPRPSDBkmdmsE2qohZHF?cluster=devnet)
and here is [2022 token deposit](https://explorer.solana.com/tx/58hDuHgDq735iv2KJoAj9Y5KaYUWzW1iaKZKZGTVin7eXWQNneu6WCEYDxozvSMRwHJ3yR9HTbkk1PZLNK7qau98?cluster=devnet) with enabled CPI guards